src/include/btree_inline.h

```diff
@@ -1525,25 +1557,38 @@ __wt_ref_block_free(WT_SESSION_IMPL *session, WT_REF *ref)
 
 /*
  * __wt_page_del_active --
- *     Return if a truncate operation is active.
+ *     Return if a truncate operation is active. "active" means approximately that the truncate is
+ *     still in progress, that is, that the underlying original page may still be required. This
+ *     function in practice is actually a visibility test (it returns whether the truncate is *not*
+ *     visible) and should be renamed and have its sense flipped to be more consistent with the rest
+ *     of the system.
  */
 static inline bool
 __wt_page_del_active(WT_SESSION_IMPL *session, WT_REF *ref, bool visible_all)
 {
     WT_PAGE_DELETED *page_del;
     uint8_t prepare_state;
 
+    /*
+     * Our caller should have already locked the WT_REF and confirmed that the previous state was
+     * WT_REF_DELETED. Consequently there are two possible cases: either ft_info.del is NULL (in
+     * which case the page is firmly deleted and no longer accessible, so the deletion is always
+     * visible) or it is not, in which case the information in ft_info.del gives us the visibility.
```

Comment:
Maybe simplify "(in which case the page is firmly deleted and no longer accessible, so the deletion is always visible)" ->

"(if the delete is globally visible and can never be rolled back)"

Reply:
Fixed in [42d20ae](https://github.com/wiredtiger/wiredtiger/pull/7909/commits/42d20ae234f880c06e2cc2133a5c772b2c7414a6).